### PR TITLE
Handle Global Cluster Nuke Failure

### PR DESCRIPTION
**Pull request by hongil0316**
## Description

Fixes https://github.com/gruntwork-io/cloud-nuke/issues/597.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.
- [ ] Attention Grunts - if this PR adds support for a new resource, ensure the `nuke_sandbox` and `nuke_phxdevops` jobs in `.circleci/config.yml` have been updated with appropriate exclusions (either directly in the job or via the `.circleci/nuke_config.yml` file) to prevent nuking IAM roles, groups, resources, etc that are important for the test accounts.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

File: aws/resources/rds_global_cluster.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	for i := 0; i < 90; i++ {
+		_, err := instance.Client.DescribeGlobalClusters(&rds.DescribeGlobalClustersInput{
+			GlobalClusterIdentifier: &name,
+		})
+		if err != nil {
+			if awsErr, isAwsErr := err.(awserr.Error); isAwsErr && awsErr.Code() == rds.ErrCodeGlobalClusterNotFoundFault {
+				return nil
+			}
+
+			return err
```
Comment (denis256) [inline aws/resources/rds_global_cluster.go:93]: make sense to use `errors.WithStackTrace(err)`?

File: aws/resources/rds_global_cluster_membership.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	if len(gdbcs.GlobalClusters) != 1 || *gdbcs.GlobalClusters[0].GlobalClusterIdentifier != name {
+		return deleted, fmt.Errorf("unexpected describe result global cluster")
+	}
+	gdbc := gdbcs.GlobalClusters[0]
+
+	deletedNames := []string{}
+	for _, member := range gdbc.GlobalClusterMembers {
+		region := strings.Split(*member.DBClusterArn, ":")[3]
+		if instance.Region != "" && instance.Region != region {
+			logging.Debugf("Skip removing cluster '%s' from global cluster since it is in diferent region", *member.DBClusterArn)
```
Comment (denis256) [inline aws/resources/rds_global_cluster_membership.go:92]: `different`

File: aws/resources/rds_global_cluster_membership.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+func (instance *DBGlobalClusterMemberships) waitUntilRdsClusterRemovedFromGlobalCluster(arnGlobalCluster string, arnCluster string) error {
+	// wait up to 15 minutes
+	for i := 0; i < 90; i++ {
+		gcs, err := instance.Client.DescribeGlobalClusters(&rds.DescribeGlobalClustersInput{
+			GlobalClusterIdentifier: &arnGlobalCluster,
+		})
+		if err != nil {
+			return err
```
Comment (denis256) [inline aws/resources/rds_global_cluster_membership.go:123]: make sense to wrap in `errors.WithStackTrace(err)`  ?

File: aws/resources/rds_global_cluster_membership.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+			return deleted, fmt.Errorf("fail to remove cluster '%s' from global cluster :%w", name, err)
+		}
+	}
+
+	return len(deletedNames) > 0, nil
+}
+
+func (instance *DBGlobalClusterMemberships) waitUntilRdsClusterRemovedFromGlobalCluster(arnGlobalCluster string, arnCluster string) error {
+	// wait up to 15 minutes
+	for i := 0; i < 90; i++ {
```
Comment (denis256) [inline aws/resources/rds_global_cluster_membership.go:118]: make sense to extract hardcoded value as constants?  (same thing for count and wait time)

**Comment by hongil0316** [comment]
Hi @denis256, addressed your comments. PTAL.

**Review by denis256 — APPROVED** [review]

**Pull request merged**